posthog/mixins.py

```diff
@@ -1,6 +1,126 @@
-from rest_framework import response, status
+import copy
+from datetime import datetime
+from itertools import zip_longest
+from typing import Any, Dict, Iterable, List, Tuple, Union
+
+import structlog
+from rest_framework import response, serializers, status
 
 from posthog.event_usage import report_user_action
+from posthog.helpers.item_history import compute_history
+from posthog.models import HistoricalVersion
+
+logger = structlog.get_logger(__name__)
+
+
+# from https://stackoverflow.com/a/4628446/222163
+def pairwise(t: List[HistoricalVersion]) -> Iterable[Tuple[HistoricalVersion, Union[HistoricalVersion, None]]]:
+    left = iter(t)
+    right = iter(t[1:])
+    return zip_longest(left, right)
+
+
+def _version_for_deletion(instance, item_id: int, team_id: int, metadata: Dict, user: Dict) -> None:
```

Comment:
We're dumping a bunch of very specific logic into a generic `posthog/mixins.py` file. This in turn means that it's now hard to read anything else in this file.

Given everything else here has to do with history: `posthog/models/history_logging.py`?